server/aesdsocket.c

```diff
@@ -0,0 +1,456 @@
+            }
+        }
+
+    close_client:
+        if (client_fd > 0)
+        {
+            close(client_fd);
+            printf("Connection Closed from %s\n", client_addr_str);
+            syslog(LOG_INFO, "Closed connection from %s", client_addr_str);
+        }
+    }
+
+    exit_cleanup();
+    return ret_status;
+}
+
+/**
+ * @brief   The function process the data read from the client. It mallocs buff_len
+ *          size and copy data from the read buffer. If '\n' is found in the buffer
+ *          then entire string will be copied to the file.
+ * 
+ * @param   
+ *  *buffer     A char array contains data read from the client
+ *  buff_len    Number of bytes present in the *buffer array.
+ * 
+ * @return  Returns number of bytes written to the file on success. Returns 0 if no
+ *          bytes were written to the file. Returns -1 when malloc fails.
+*/
+int process_read_data(char *buffer, int buff_len)
+{
+    static char *malloc_buffer = NULL;
+    static int malloc_buffer_len = 0;
+    int index = 0;
+
+    // Find the delimeter '\n' in the received buffer
+    for (index = 0; index < buff_len && buffer[index] != '\n'; index++)
+        ;
+
+    if (malloc_buffer == NULL)
+        malloc_buffer = (char *)malloc(sizeof(char) * (index == buff_len ? buff_len : index + 1));
+    else
+        malloc_buffer = (char *)realloc(malloc_buffer, (malloc_buffer_len + (index == buff_len ? buff_len : index + 1)));
+
+    if (malloc_buffer == NULL)
+    {
+        return -1;
+    }
+
+    // copy data including \n and update malloc buffer length
+    memcpy(malloc_buffer + malloc_buffer_len, buffer, (index == buff_len ? buff_len : index + 1));
+    malloc_buffer_len += (index == buff_len ? buff_len : index + 1);
+
+    int ret_len = malloc_buffer_len;
+
+    if (index < buff_len)
+    {
+        if (write(file_fd, malloc_buffer, malloc_buffer_len) < 0)
+        {
+            printf("Error: %s : Detected error while writing to the file\n", strerror(errno));
+            syslog(LOG_ERR, "Error: %s : Detected error while writing to the file\n", strerror(errno));
+            ret_len = 0;
+        }
+        free(malloc_buffer);
+        malloc_buffer = NULL;
+        malloc_buffer_len = 0;
+
+        return ret_len;
+    }
+
+    return 0;
+}
+
+/**
+ * @brief   The function process the data read from the client. It mallocs buff_len
+ *          size and copy data from the read buffer. If '\n' is found in the buffer
+ *          then entire string will be copied to the file.
+ * 
+ * @param   
+ *  *buffer     A char array contains data read from the client
+ *  buff_len    Number of bytes present in the *buffer array.
+ * 
+ * @return  Returns number of bytes written to the file on success. Returns -1 when 
+ *          malloc fails.
+*/
+int get_write_data(char **malloc_buffer, int total_size, int file_fd)
+{
+    if (*malloc_buffer != NULL)
+        free(*malloc_buffer);
+
+    *malloc_buffer = (char *)malloc(sizeof(char) * total_size);
```

Comment:
@ajaykandagal what assumption are you making here about the size of the process heap and the total size of all packets sent?